### ISSUE TITLE
Zero-value of ClientOptions is invalid

### ISSUE THREAD

**Issue by abhinav** [uber/cherami-client-go#13]
For an API like [NewClient](https://godoc.org/github.com/uber/cherami-client-go/client/cherami#NewClient) which accepts [ClientOptions](https://godoc.org/github.com/uber/cherami-client-go/client/cherami#ClientOptions), the expectation is that you pass in a struct customizing only the values you want to change from the defaults and the rest can be left as-is to use the default values for them.

That is, I should be able to do any of the following:

```go
NewClient(..., &ClientOptions{})  // Equivalent to NewClient(..., nil)
NewClient(..., &ClientOptions{DeploymentStr: "dev"})
NewClient(..., &ClientOptions{MetricsReporter: reporter, Logger: myLogger})
```

For the examples above, `ClientOptions.Timeout` will be zero which is a problem because,

- `NewClient` calls [`newClientWithTChannel`](https://sourcegraph.com/github.com/uber/cherami-client-go@1cdcfc4a204fe42013f4cf7196fab208d3b98085/-/blob/client/cherami/client.go#L100:6-100:27)
- which calls [`verifyOptions`](https://sourcegraph.com/github.com/uber/cherami-client-go@1cdcfc4a204fe42013f4cf7196fab208d3b98085/-/blob/client/cherami/client.go#L101:12-101:25)
- which calls [`common.VerifyTimeout`](https://sourcegraph.com/github.com/uber/cherami-client-go@1cdcfc4a204fe42013f4cf7196fab208d3b98085/-/blob/common/util.go#L280:6-280:21)
- which **panics** if Timeout is zero

Please apply the defaults for ClientOptions for the zero-value case. Also, NewClient should not panic because of invalid options especially since it already has an `error` in its return values.

**Comment by datoug** [comment]
ack, working on it.

**Comment by venkat1109** [comment]
this is fixed as part of this PR #13
